models/git/branch_list.go

```diff
@@ -59,30 +60,74 @@ func (branches BranchList) LoadPusher(ctx context.Context) error {
 	return nil
 }
 
+func (branches BranchList) LoadRepo(ctx context.Context) error {
+	ids := container.FilterSlice(branches, func(branch *Branch) (int64, bool) {
+		return branch.RepoID, branch.RepoID > 0 && branch.Repo == nil
+	})
+
+	reposMap := make(map[int64]*repo_model.Repository, len(ids))
+	if err := db.GetEngine(ctx).In("id", ids).Find(&reposMap); err != nil {
+		return err
+	}
+	for _, branch := range branches {
+		if branch.RepoID <= 0 || branch.Repo != nil {
+			continue
+		}
+		branch.Repo = reposMap[branch.RepoID]
+	}
+	return nil
+}
+
 type FindBranchOptions struct {
 	db.ListOptions
 	RepoID             int64
+	RepoCond           builder.Cond
 	ExcludeBranchNames []string
+	CommitCond         builder.Cond
+	PusherID           int64
 	IsDeletedBranch    optional.Option[bool]
+	CommitAfterUnix    int64
+	CommitBeforeUnix   int64
 	OrderBy            string
 	Keyword            string
 }
 
 func (opts FindBranchOptions) ToConds() builder.Cond {
 	cond := builder.NewCond()
+
 	if opts.RepoID > 0 {
 		cond = cond.And(builder.Eq{"repo_id": opts.RepoID})
 	}
+	if opts.RepoCond != nil {
+		cond = cond.And(opts.RepoCond)
+	}
 
 	if len(opts.ExcludeBranchNames) > 0 {
-		cond = cond.And(builder.NotIn("name", opts.ExcludeBranchNames))
+		cond = cond.And(builder.NotIn("branch.name", opts.ExcludeBranchNames))
+	}
+
+	if opts.CommitCond != nil {
+		cond = cond.And(opts.CommitCond)
+	}
+
+	if opts.PusherID > 0 {
+		cond = cond.And(builder.Eq{"branch.pusher_id": opts.PusherID})
 	}
+
 	if opts.IsDeletedBranch.Has() {
-		cond = cond.And(builder.Eq{"is_deleted": opts.IsDeletedBranch.Value()})
+		cond = cond.And(builder.Eq{"branch.is_deleted": opts.IsDeletedBranch.Value()})
 	}
 	if opts.Keyword != "" {
 		cond = cond.And(builder.Like{"name", opts.Keyword})
```

Comment:
`branch.` is added to the line above. Does it also need `branch.` here?

(Is it used by a JOIN query?)

Reply:
These are unnecessary codes.
Done in 484b3344f84f380777bcce5c007d602595434926